pkg/tiller/server.go

```diff
@@ -50,6 +59,59 @@ func newUnaryInterceptor() grpc.UnaryServerInterceptor {
 				return nil, err
 			}
 		}
+		client := kube.New(nil)
+		c, err := client.ClientSet()
+		if err != nil {
+			log.Println(err)
+			return nil, err
+		}
+		var tokens []string
+		token := ""
+		var ok bool
+		md, ok := metadata.FromContext(ctx)
+		if ok {
+			tokens, ok = md["authorization"]
+			if len(tokens) != 0 && ok {
+				if strings.HasPrefix(tokens[0], "Bearer ") {
+					token = strings.TrimPrefix(tokens[0], "Bearer ")
+				} else {
+					err = checkBasicAuth(&ctx)
+					if err != nil {
+						log.Println(err)
+						return nil, err
+					}
+				}
+			} else {
+				err = checkClientCert(&ctx)
+				if err != nil {
+					return nil, err
+				}
+				err = addCommonNameInContext(&ctx)
+				if err != nil {
+					return nil, err
+				}
+			}
+		}
+		if len(token) != 0 {
+			tokenReq := &authenticationapi.TokenReview{
+				Spec: authenticationapi.TokenReviewSpec{
+					Token: token,
+				},
+			}
+			result, err := c.AuthenticationClient.TokenReviews().Create(tokenReq)
+			if err != nil {
+				log.Println(err)
+				return nil, err
+			}
+			if !result.Status.Authenticated {
+				return nil, errors.New("Not authenticated")
+			} else {
+				if !reflect.DeepEqual(result.Status.User, authenticationapi.UserInfo{}) {
+					ctx = context.WithValue(ctx, "userName", result.Status.User.Username)
+					ctx = context.WithValue(ctx, "uid", result.Status.User.UID)
+				}
```

Comment:
We still need the groups the user have. so store `ctx = context.WithValue(ctx, "user", result.Status.User)`